core/src/main/java/org/axonframework/commandhandling/model/GenericJpaRepository.java

```diff
@@ -154,11 +268,31 @@ public GenericJpaRepository(EntityManagerProvider entityManagerProvider, Aggrega
      */
     public GenericJpaRepository(EntityManagerProvider entityManagerProvider, Class<T> aggregateType, EventBus eventBus,
                                 LockFactory lockFactory, Function<String, ?> identifierConverter) {
+        this(entityManagerProvider, aggregateType, eventBus, null, lockFactory, identifierConverter);
```

Comment:
So it's okay for the `RepositoryProvider` to be `null` in several scenarios? How come?

I think this'll occur several times in this PR as this is already the second time, so I wont comment on any other occurrences of `RepositoryProvider == null`.